src/pykeen/triples/splitting.py

```diff
@@ -275,6 +255,194 @@ def _prepare_cleanup(
     return to_move_mask
 
 
+class RandomizedCleaner(Cleaner):
+    """Cleanup a triples array by randomly selecting testing triples and recalculate to minimize moves.
+
+    1. Calculate ``move_id_mask`` as in :func:`_prepare_cleanup`
+    2. Choose a triple to move, recalculate ``move_id_mask``
+    3. Continue until ``move_id_mask`` has no true bits
+    """
+
+    def cleanup_pair(
+        self,
+        reference: MappedTriples,
+        other: MappedTriples,
+        random_state: torch.Generator,
+    ) -> Tuple[MappedTriples, MappedTriples]:  # noqa: D102
+        generator = ensure_torch_random_state(random_state)
```

Comment:
is this necessary, since the type signature says a proper generator will always be passed?

Reply:
ff4b499